File: Problem Set 6/sentimental-readability/readability.py
text = input("Text: ")
letters = 0
words = 1
sentences = 0
# Iterate through each character in the text
for char in text:
    # Check if character is in the alphabet
    if char.isalpha():
        letters += 1
    elif char == " ":
        words += 1
    elif char == "." or char == "!" or char == "?":
        sentences += 1
letters = round(letters / words * 100, 2)
sentences = round(sentences / words * 100, 2)
index = round(0.0588 * letters - 0.296 * sentences - 15.8)
if index >= 16:
    print("Grade 16+")
elif index < 1:
    print("Before Grade 1")
else:
    print(f"Grade {index}")

File: Problem Set 2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>
#include <ctype.h>

int main(void)
{
    // Get the text from the user
    string text = get_string("Text: ");
    // Get number of letters in text
    int characters = strlen(text);
    float letters = 0;
    float words = 1;
    float sentences = 0;
    for (int i = 0; i < characters; i++){
        if (isupper(text[i])){
            letters++;
        }
        if (islower(text[i])){
            letters++;
        }
        //Get number of words in text
        if (text[i] == ' '){
            words++;
        }
        //Get number of sentences in text
        if (text[i] == '.' || text[i] == '!' || text[i] == '?'){
            sentences++;
        }
    }
    // Coleman-Liau index
    int index = round((0.0588 * letters / words * 100) - (0.296 * sentences / words * 100) - 15.8);
    if (index >= 1 && index <= 16){
        printf("Grade %i\n", index);
    }
    if (index < 1){
        printf("Before Grade 1\n");
    }
    if (index > 16){
        printf("Grade 16+\n");
    }
}